tests/waku_lightpush/test_client.nim

```diff
@@ -0,0 +1,277 @@
+
+      # When publishing a valid payload
+      handlerFuture = newPushHandlerFuture()
+      let publishResponse8 = await client.publish(pubsub_topics.LEGACY_ENCODING, message8, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      discard await handlerFuture.withTimeout(FUTURE_TIMEOUT)
+      check:
+        publishResponse8.isOk()
+        handlerFuture.finished()
+      
+      # And the message is received with the correct topic and payload
+      check (pubsub_topics.LEGACY_ENCODING, message8) == handlerFuture.read()
+
+      # When publishing a valid payload
+      handlerFuture = newPushHandlerFuture()
+      let publishResponse9 = await client.publish(pubsubTopic, message9, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      discard await handlerFuture.withTimeout(FUTURE_TIMEOUT)
+      check:
+        publishResponse9.isOk()
+        handlerFuture.finished()
+      
+      # And the message is received with the correct topic and payload
+      check (pubsubTopic, message9) == handlerFuture.read()
+
+    xasyncTest "Valid Paylod Sizes":
+      # Given some valid payloads
+      let
+        message1 = fakeWakuMessage(contentTopic=contentTopic, payload=getByteSequence(1024)) # 1KiB
+        message2 = fakeWakuMessage(contentTopic=contentTopic, payload=getByteSequence(10*1024)) # 10KiB 
+        message3 = fakeWakuMessage(contentTopic=contentTopic, payload=getByteSequence(100*1024)) # 100KiB
+        message4 = fakeWakuMessage(contentTopic=contentTopic, payload=getByteSequence(1*1024*1024 + 63 * 1024 + 911)) # 1113999B -> Max Size (Inclusive Limit)
+        message5 = fakeWakuMessage(contentTopic=contentTopic, payload=getByteSequence(1*1024*1024 + 63 * 1024 + 912)) # 1114000B -> Max Size (Exclusive Limit)
+      
+      # When publishing the 1KiB payload
+      let publishResponse1 = await client.publish(pubsubTopic, message1, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      check:
+        publishResponse1.isOk()
+        (pubsubTopic, message1) == (await handlerFuture.waitForResult()).value()
+
+      # When publishing the 10KiB payload
+      handlerFuture = newPushHandlerFuture()
+      let publishResponse2 = await client.publish(pubsubTopic, message2, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      check:
+        publishResponse2.isOk()
+        (pubsubTopic, message2) == (await handlerFuture.waitForResult()).value()
+
+      # When publishing the 100KiB payload
+      handlerFuture = newPushHandlerFuture()
+      let publishResponse3 = await client.publish(pubsubTopic, message3, serverRemotePeerInfo)
+
+      # Then the message is received by the server
+      check:
+        publishResponse3.isOk()
+        (pubsubTopic, message3) == (await handlerFuture.waitForResult()).value()
+
+      # When publishing the 1MiB + 63KiB + 911B payload (1113999B)
```

Comment:
This triggers SEGFAULT.

Reply:
Interesting, we need to debug why this happens, although it seems some kind of overflow

Reply:
Indeed, this has already been reported here: https://github.com/waku-org/nwaku/issues/2271

Reply:
what's the status with the segfault? Is the PR ready or are we still waiting for this part?

Reply:
That's been fixed. Apparently I forgot to change the comments (I thought I had fixed them), but it's working thanks to a fix by @Ivansete-status